### PR TITLE
Task1 core concepts 

### PR DESCRIPTION
**Pull request by maxxbay**
Header

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+  const genres = [
+    'ALL',
+    'Action',
+    'Adventure',
+    'Comedy',
+    'Drama',
+    'Horror',
+    'Sci-Fi',
+  ];
+  const [, setSearchValue] = useState('');
```
Comment (lenaricabal) [inline src/App.jsx:16]: why do you need this state here?

File: src/components/GenreSelect.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React, { useState, useCallback } from 'react';
+import './GenreSelect.scss';
+
+const GenreSelect = ({ genres }) => {
+  const [genre, setGenre] = useState('');
+
+  // console.log(genres);
+  return (
+    <div className="genre-select">
+      <select
```
Comment (lenaricabal) [inline src/components/GenreSelect.js:10]: if you will create your own select, you'll be able to style it like on design

File: src/components/SearchForm.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { useState, useCallback } from 'react';
+import ButtonSearch from './Button/Button';
+import './SearchForm.scss';
+
+const SearchForm = ({ setSearchValue }) => {
```
Comment (lenaricabal) [inline src/components/SearchForm.js:5]: search form should accept 2 arguments 
```
The component should accept two properties:

1. Initial search query. Use the value to set the initial value for the input
2. A "onSearch" callback property. Call the callback property every time the user presses Enter when the input has focus or when the user clicks the Search button. Pass current input value in callback arguments.

```

File: src/components/GenreSelect.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React, { useState, useCallback } from 'react';
+import './GenreSelect.scss';
+
+const GenreSelect = ({ genres }) => {
```
Comment (lenaricabal) [inline src/components/GenreSelect.js:4]: this component should accept three props:
```
The component should accept three properties:

A list of genre names to display. Use the incoming list to render genre buttons.
A name of currently selected genre. Use the name to identify which button to highlight.
A "onSelect" callback property. Call the callback function when the user clicks on any genre button. Pass respective genre name to the callback arguments.
```

File: src/components/Header/Header.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import NetflixLogo from 'common/Logo/NetflixLogo';
+import './Header.scss';
+import SearchForm from 'components/SearchForm';
+
+const Header = ({ setSearchValue }) => {
+  return (
+    <div className="header">
```
Comment (lenaricabal) [inline src/components/Header/Header.js:7]: it's better to use semantic <header> tag

File: src/components/SearchForm.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    },
+    [value, setSearchValue]
+  );
+
+  const handleChange = useCallback(({ target: { value } }) => {
+    setValue(value);
+  }, []);
+
+  return (
+    <div className="search" onSubmit={handleForm}>
```
Comment (lenaricabal) [inline src/components/SearchForm.js:21]: it is an extra `onSubmit`

File: src/components/Button/Button.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import './Button.scss';
+
+const ButtonSearch = ({ children, type }) => {
```
Comment (lenaricabal) [inline src/components/Button/Button.js:4]: need to change name to `Button`

File: src/App.jsx [hunk 8]
```diff
@@ -1,11 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
+import './App.scss';
+import GenreSelect from 'components/GenreSelect';
+import Header from './components/Header/Header';
+
+const App = () => {
+  const genres = [
```
[the remark this answers is not in the document]
Comment (maxxbay) [reply src/App.jsx:7]: done

File: src/App.jsx (hunk 1 above)
Comment (maxxbay) [reply src/App.jsx:16]: My mistake

File: src/components/SearchForm.js (hunk 3 above)
Comment (maxxbay) [reply src/components/SearchForm.js:5]: done

File: src/components/Header/Header.js (hunk 5 above)
Comment (maxxbay) [reply src/components/Header/Header.js:7]: done

File: src/components/SearchForm.js (hunk 6 above)
Comment (maxxbay) [reply src/components/SearchForm.js:21]: done

File: src/components/Button/Button.js (hunk 7 above)
Comment (maxxbay) [reply src/components/Button/Button.js:4]: done

**Pull request closed without merging**